### PR TITLE
[#6748][sql] Fix is misleading in presto configuration

### LINKED ISSUE #6748: `pulsar.broker-service-url` is misleading in presto configuration
*Motivation*

`pulsar.broker-service-url` in presto configuration is misleading. The name seems to indicate that you should use a broker service URL. But actually it is expecting the admin/HTTP service URL. So it is better to add a new `pulsar.admin-service-url` to replace this `pulsar.broker-service-url` setting.

At least, we need to document this setting.

### PR DESCRIPTION
**Pull request by eaba**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #6748

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: conf/presto/catalog/pulsar.properties [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # name of the connector to be displayed in the catalog
 connector.name=pulsar
 # the url of Pulsar broker service
-pulsar.broker-service-url=http://localhost:8080
+pulsar.web-service-url=http://localhost:8080
```
Comment (sijie) [inline conf/presto/catalog/pulsar.properties:23]: @eaba I don't think we should rename this setting. We should introduce a new setting and deprecate the old setting.

Comment (eaba) [reply conf/presto/catalog/pulsar.properties:23]: `pulsar.web-service-url` as a new setting?

**Comment by sijie** [comment]
@eaba yes. keep the old setting and add a new setting. If the new setting is set, use the new setting; otherwise, fall back to use the old setting.

File: conf/presto/catalog/pulsar.properties [hunk 2]
```diff
@@ -20,7 +20,8 @@
 # name of the connector to be displayed in the catalog
 connector.name=pulsar
 # the url of Pulsar broker service
-pulsar.broker-service-url=http://localhost:8080
+pulsar.broker-service-url=http://localhost:8080 # DEPRECATED
```
Comment (codelipenghui) [inline conf/presto/catalog/pulsar.properties:23]: It's better to move ` # DEPRECATED` to the above line.

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarConnectorConfig.java [hunk 3]
```diff
@@ -76,14 +77,24 @@
...
     }
-
-    @Config("pulsar.broker-service-url")
+   @Config("pulsar.broker-service-url")
     public PulsarConnectorConfig setBrokerServiceUrl(String brokerServiceUrl) {
         this.brokerServiceUrl = brokerServiceUrl;
         return this;
     }
+    @Config("pulsar.web-service-url")
+    public PulsarConnectorConfig setBrokerServiceUrl(String webServiceUrl) {
```
Comment (codelipenghui) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarConnectorConfig.java:94]:
```suggestion
    public PulsarConnectorConfig setWebServiceUrl(String webServiceUrl) {
```

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarConnectorConfig.java [hunk 4]
```diff
@@ -76,14 +77,24 @@
...
     @NotNull
     public String getBrokerServiceUrl() {
-        return brokerServiceUrl;
+		if(StringUtils.isEmpty(webServiceUrl))
+        { 
+		  return brokerServiceUrl;
+        }
+        else{
+            return webServiceUrl;
+        }
```
Comment (codelipenghui) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarConnectorConfig.java:86]: Please format the code.

File: site2/website/versioned_docs/version-2.2.0/sql-deployment-configurations.md [hunk 5]
```diff
@@ -16,7 +16,7 @@ The configurations for the connector and its default values are discribed below.
 connector.name=pulsar
 
 # the url of Pulsar broker service
-pulsar.broker-service-url=http://localhost:8080
+pulsar.web-service-url=http://localhost:8080
```
Comment (codelipenghui) [inline site2/website/versioned_docs/version-2.2.0/sql-deployment-configurations.md:19]: I think we don't need to change the documentation of 2.2.0 version right?

Comment (eaba) [reply site2/website/versioned_docs/version-2.2.0/sql-deployment-configurations.md:19]: Sure

**Comment by codelipenghui** [comment]
@eaba Sorry for the late response, could you please help take a look at the CI problems? Looks related to this change. Thanks.

@sijie Please help review this PR again.

**Comment by eaba** [comment]
@codelipenghui I don't think the failing tests has to do with this PR

**Comment by jiazhai** [comment]
@eaba looks like the failure is related to this change, Would you please help check it?
https://github.com/apache/pulsar/pull/6757/checks?check_run_id=657446803
```
20:02:02.527 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@216] INFO  org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): Executing...
20:02:03.740 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@221] INFO  org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): STDERR: Query
20:02:03.740 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@221] INFO  org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): STDERR: 20200508_200203_00007_z87n8
20:02:03.742 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@221] INFO  org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): STDERR: failed:
20:02:03.742 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@221] INFO  org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): STDERR: Failed to get schemas from pulsar: Connection refused
20:02:03.745 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@221] INFO  org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): STDERR: 
20:02:03.745 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@221] INFO  org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): STDERR: 
20:02:03.779 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@236] INFO  org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): Done
20:02:03.789 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@254] INFO  org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): completed with 1
20:02:03.789 [tc-okhttp-stream-64606043:org.apache.pulsar.tests.integration.utils.DockerUtils$2@257] ERROR org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(TestBasicPresto-ydcsr-presto-worker:/bin/bash -c /pulsar/bin/pulsar sql --execute 'show schemas in pulsar;'): completed with non zero return code: 1
stdout: 
stderr: Query 20200508_200203_00007_z87n8 failed: Failed to get schemas from pulsar: Connection refused
```

**Comment by eaba** [comment]
How does the `pulsar.properties` get injected?
How about this?:
````
public void startPrestoWorker() {
        if (null == prestoWorkerContainer) {
            prestoWorkerContainer = new PrestoWorkerContainer(clusterName, PrestoWorkerContainer.NAME)
                    .withNetwork(network)
                    .withNetworkAliases(PrestoWorkerContainer.NAME)
                    .withEnv("clusterName", clusterName)
                    .withEnv("zkServers", ZKContainer.NAME)
                    .withEnv("zookeeperServers", ZKContainer.NAME + ":" + ZKContainer.ZK_PORT)
                    .withEnv("pulsar.zookeeper-uri", ZKContainer.NAME + ":" + ZKContainer.ZK_PORT)
                    .withEnv("pulsar.broker-service-url", "http://pulsar-broker-0:8080");
        }
        log.info("Starting Presto Worker");
        prestoWorkerContainer.start();
    }
````

**Comment by jiazhai** [comment]
right, this should work.

> How does the `pulsar.properties` get injected?
> How about this?:
> 
> ```
> public void startPrestoWorker() {
>         if (null == prestoWorkerContainer) {
>             prestoWorkerContainer = new PrestoWorkerContainer(clusterName, PrestoWorkerContainer.NAME)
>                     .withNetwork(network)
>                     .withNetworkAliases(PrestoWorkerContainer.NAME)
>                     .withEnv("clusterName", clusterName)
>                     .withEnv("zkServers", ZKContainer.NAME)
>                     .withEnv("zookeeperServers", ZKContainer.NAME + ":" + ZKContainer.ZK_PORT)
>                     .withEnv("pulsar.zookeeper-uri", ZKContainer.NAME + ":" + ZKContainer.ZK_PORT)
>                     .withEnv("pulsar.broker-service-url", "http://pulsar-broker-0:8080");
>         }
>         log.info("Starting Presto Worker");
>         prestoWorkerContainer.start();
>     }
> ```
> 
> This has to do with connection to either pulsar broker or pulsar sql

**Comment by jiazhai** [comment]
> How does the `pulsar.properties` get injected?
> How about this?:
> 
> ```
> public void startPrestoWorker() {
>         if (null == prestoWorkerContainer) {
>             prestoWorkerContainer = new PrestoWorkerContainer(clusterName, PrestoWorkerContainer.NAME)
>                     .withNetwork(network)
>                     .withNetworkAliases(PrestoWorkerContainer.NAME)
>                     .withEnv("clusterName", clusterName)
>                     .withEnv("zkServers", ZKContainer.NAME)
>                     .withEnv("zookeeperServers", ZKContainer.NAME + ":" + ZKContainer.ZK_PORT)
>                     .withEnv("pulsar.zookeeper-uri", ZKContainer.NAME + ":" + ZKContainer.ZK_PORT)
>                     .withEnv("pulsar.broker-service-url", "http://pulsar-broker-0:8080");
>         }
>         log.info("Starting Presto Worker");
>         prestoWorkerContainer.start();
>     }
> ```
> 
> This has to do with connection to either pulsar broker or pulsar sql

@eaba right, this should work.

**Comment by jiazhai** [comment]
And this is a enhancement than a bugfix, would like to remove it from 2.5.2, if it is really needed, we could add it back.

**Comment by codelipenghui** [comment]
move to 2.7.0 first.

**Comment by codelipenghui** [comment]
@eaba Could you please help resolve the conflicts?

**Comment by Jennifer88huang** [comment]
Since it's fixed in https://github.com/apache/pulsar/pull/8549, I assume we can close this PR.

**Pull request closed without merging**